### PR TITLE
Add selection methods, tests, and cleanup

### PR DESCRIPTION
**Pull request by edenh**
Ref #154, #157 

- adds `Draw.select`, `Draw.selectAll`, `Draw.deselect`, and `Draw.deselectAll`
- renames `Draw.set` to `Draw.add`
- tests
- whitespace cleanup

### CODE REVIEW

**Comment by mcwhittemore** [comment]
#157 is maybe too philosophical, so I'm going to use this PR to help bring my thoughts down a bit.

`Draw.select` and `Draw.selectAll` moves features from the non editing to the editing state. This is currently done by gl-draw when a users clicks on the feature. Key bindings being part of gl-draw already, you can assume `cmd+a` would perform a `Draw.selectAll()` internally.

While these two use cases are the obvious ones, you could also imagine an edit button in an app that would use `Draw.select` to start editing one feature.

You can also imagine a user wanting to highlight a set of features and using a bunch of `Draw.select` calls to put the features into edit mode which changes their style.

For `Draw.unselect` you can imagine a user wanting to stop editing when their user clicks on a different tab in their application.

One problem I see here is that this muddles the concept of selected with editable. I think we will want to add vertex selection in time. This would allow users to grab a few vertices and move them at the same time. Maybe this is something we should ignore until we get there? Worse case it would be a breaking change...

Also, the use case of using selectAll to show highlight a feature by putting it into edit mode, sounds like a request to style features using more than the two default buckets of gl-draw.

All of that said, we're still in 0.* so lets just do stuff and refactor later when we have more than theoretical other options.

**Comment by mcwhittemore** [comment]
We should introduce a changelog and mention that this is a breaking change.

File: src/api.js [hunk 1]
```diff
@@ -106,15 +106,57 @@ export default class API extends mapboxgl.Control {
...
+    return this;
+  }
+
+  /**
+   * deselect all features for editing
+   * @param {String} id - the drawId of the feature
+   * @returns {Draw} this
+   */
+  deselectAll() {
+    this._store.getAllIds()
```
Comment (mcwhittemore) [inline src/api.js]: Lets filter this first, just in case the unneeded calls to commit make more renders and slow things down.

File: src/api.js [hunk 2]
```diff
@@ -106,15 +106,57 @@ export default class API extends mapboxgl.Control {
...
+      .forEach(id => this.commit(id));
+    return this;
+  }
+
+  /**
+   * get a feature collection of features being edited
    * @return {Object} a feature collection of geometries that are in edit mode
    */
-  getEditing() {
+  getSelected() {
```
Comment (mcwhittemore) [inline src/api.js]: What are you using this for? Seems weird to me to change the api. Are you looking for something to give you the list of feature ids that are in edit mode?

File: src/store.js [hunk 3]
```diff
@@ -86,21 +86,21 @@ export default class Store {
    */
   clearEditing() {
```
Comment (mcwhittemore) [inline src/store.js]: This should be changed to delete too.

File: src/api.js (hunk 1 above)
Comment (mcwhittemore) [inline src/api.js]: `_store.getEditIds.forEach` would be better even

**Comment by mcwhittemore** [comment]
@edenh - this is looking good. I think we should push this a bit more and just get rid of the concept of editing. This would help kill some of my fears from the long post above and I think would help make the code a bit cleaner.

File: src/api.js (hunk 1 above)
Comment (edenh) [inline src/api.js]: I think it should commit. UX-wise, I would assume deselecting does not "revert" my changes.

**Comment by edenh** [comment]
Updated to use `selected` terminology over `edit`, as well as a changelog.

**Comment by mcwhittemore** [comment]
Looks good, but I don't see the changelog

**Comment by edenh** [comment]
Added here https://github.com/mapbox/mapbox-gl-draw/commit/6827ebe3f512c680234006fde4ec649623fba825

File: src/api.js [hunk 4]
```diff
@@ -13,14 +13,14 @@ export default class API extends mapboxgl.Control {
...
-   * @pata {Boolean} [options.permanent=false] - disable editing for feature
+   * @pata {Boolean} [options.permanent=false] - disable selection for feature
    * @returns {Number} draw id of the set feature or an array of
    * draw ids if a feature collection was added
    */
-  set(feature, options) {
+  add(feature, options) {
     feature = JSON.parse(JSON.stringify(feature));
     if (feature.type === 'FeatureCollection') {
       return feature.features.map(subFeature => this.set(subFeature, options));
```
Comment (kelvinabrokwa) [inline src/api.js]: this should also be changed to `this.add`

**Comment by kelvinabrokwa** [comment]
we should also be firing `draw.add` if we've changed the naming scheme

https://github.com/mapbox/mapbox-gl-draw/blob/master/src/store.js#L32-L35
https://github.com/mapbox/mapbox-gl-draw/blob/master/src/store.js#L132-L135

**Pull request merged**